identify

```diff
@@ -2,8 +2,36 @@
 cd "$(dirname "$0")"
 if [[ $# != 1 ]]; then
   echo >&2 "Usage: $0 path/to/libc.so"
+  echo >&2 "    OR $0 bid=<BUILD_ID>"
+  echo >&2 "    OR $0 md5=<MD5>"
+  echo >&2 "    OR $0 sha1=<SHA1>"
+  echo >&2 "    OR $0 sha256=<SHA256>"
   exit 2
 fi
-libc=$1
-sha1=`sha1sum $libc | awk '{print $1}'`
-sha1sum db/*.so | grep "$sha1 " | perl -n -e '/db\/(.*)\.so/&&print "id $1\n"'
+
+arg="$1"
+
+if [[ -f "$arg" ]]; then
+  libc=$arg
+  arg="sha1=$(sha1sum "$libc" | awk '{print $1}')"
+fi
+
+if [[ "$arg" == "bid="* ]] ; then
+  hash="${arg#"bid="}"
+  file db/*.so | grep "=$hash," | perl -n -e '/db\/(.*)\.so:.*/&&print "id $1\n"'
```

Comment:
should we use xargs here? what is the command line limit these days?

Reply:
Yeah, `ls | xargs` is so much faster ^^
Actually, our files have only alphanumeric filename, thus, no need to use find (https://github.com/koalaman/shellcheck/wiki/SC2011)